[Backport release-25.05] stdenv: Add CPE fields to meta

YorikSar opened the pull request:
Backport to `release-25.05`, done manually because of failed automatic backport by a label in #409797.

* [ ]  Before merging, ensure that this backport is [acceptable for the release](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#changes-acceptable-for-releases).
  
  * Even as a non-committer, if you find that it is not acceptable, leave a comment.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

infinisil commented:
@YorikSar Based on the shown diff it doesn't look right :thinking:

K900 reviewed (changes requested):
https://github.com/NixOS/nixpkgs/pull/409797#issuecomment-3238995897 needs addressing first.

YorikSar committed:
Reapply "stdenv: Add CPE fields to meta"

This reverts commit de74f9caf0854a9ee8c39774cf4521870fa1f44a.

(cherry picked from commit dd12290517ff6e57ab2ca384b101e0cdc617570c)

YorikSar committed:
stdenv: Remove all null values from meta.identifiers

nix-env writes a warning for each derivation that has null in its meta
values, so fields without known values are removed from the result.

Fixes issue raised by @K900 in https://github.com/NixOS/nixpkgs/pull/409797#issuecomment-3238995897

(cherry picked from commit a178fd8c436915d233a1de37a2b211489aad5340)

YorikSar committed:
stdenv: Address comments by @ConnorBaker

(cherry picked from commit 5e1eee582748866a51ca8f780829c3b2cf523fc3)
